Merge from personal repository #5

GalaticStryder opened the pull request:
Already have a better solution for that, shouldn't be shipped with the
ROM because it doesn't match the design.

Signed-off-by: Ícaro Hoff <icarohoff@gmail.com>

GalaticStryder committed:
zl1: remove proprietary IR controller

Already have a better solution for that, shouldn't be shipped with the
ROM because it doesn't match the design.

Signed-off-by: Ícaro Hoff <icarohoff@gmail.com>

TheCrazyLex commented:
Can you bundle the perfd update with this? Probably also bundle the changelog of the current binary you have in the commit message of that, as you missed that

TheCrazyLex committed:
zl1: Update our custom perfd from upstream

Thanks to Alex for improving perfd routines for SD821 on oneplus3,
this is just a binary importation, nothing has been changed nor modified.

Current: 487a70dd213bd70653f4e4bdf50c028c11ad0b45
Updated: 138fdd0b2964fde0292e9914a5b12b7a5228153e

Changelog:

oneplus3: Update perfd to improve latency | 138fdd0b2964fde0292e9914a5b12b7a5228153e

Update perfd with a perflock latency improvement.

oneplus3: Update performance stack | 487a70dd213bd70653f4e4bdf50c028c11ad0b45

Apply changes for conditional frequency limiting
in low-performance cases.

oneplus3: minor update for perfd | 9b0526dbd4c82aa2cee0213dd3a9b18dec35243f

* Adjust logging.
* Rebuilt in 7.1.

oneplus3: update perfd with some changes | 575cc2e71341183793bcff7e5af92507c6fec915

Those changes are mostly needed due to mistakes
Qualcomm made when designing mp-ctl (perfd) for
msm8996pro.

oneplus3: perf: apply some of my changes | ff55e3389db5b6b9afea73fdede3ca10b616bae9

init: set proper min freqs for msm8996
mp-ctl: rework initialisation semantics
mp-ctl: fix major race with init sequence and mpctl_send
mp-ctl: Secure request termination operations
mp-ctl: Do self destruction if we fail before stage 4
client: log QLOGE on non-debugging builds

oneplus3: Magic perf modules | 5b9814d1029745ab4c96a182278b17c406d3fa27

This commit patches the perf modules which
we have already compiled with our routines
and compile time improvements.

Upstream: https://github.com/ThankYouMario/proprietary_vendor_oneplus

Signed-off-by: Alex Naidis <alex.naidis@linux.com>
Signed-off-by: Ícaro Hoff <icarohoff@gmail.com>
